references/identity-facade/pipeline.sh

```diff
@@ -0,0 +1,356 @@
+
+    # discovery document of an OIDC compliant IdP
+    idp_discovery_document=https://"$APIGEE_ORG"-"$APIGEE_ENV".apigee.net/v1/openid-connect/.well-known/openid-configuration
+
+    # retrieve configuration data from a discovery document
+    response=$(curl --silent -k1 -fsSL -X GET -H "Accept:application/json" "$idp_discovery_document")
+    if [ "$( printf '%s' "$response" | grep -c error )" -ne 0  ]; then
+        echo "$response"
+        
+        exit 1
+    fi
+
+    # extract data used to feed the kvm
+    issuer=$( printf '%s' "$response" | jq .issuer )
+    authorization_endpoint=$( printf '%s' "$response" | jq .authorization_endpoint )
+    token_endpoint=$( printf '%s' "$response" | jq .token_endpoint )
+    jwks_uri=$( printf '%s' "$response" | jq .jwks_uri )
+    userinfo_endpoint=$( printf '%s' "$response" | jq .userinfo_endpoint )
+
+    # set env variables for kvm (idpConfig)
+    TEST_IDP_ISSUER=$(printf '%s' "$issuer" | awk -F\" '{print $2}' | awk -F\" '{print $1}')
+    export TEST_IDP_ISSUER
+    
+    TEST_IDP_APIGEE_REDIRECT_URI="https://$APIGEE_ORG-$APIGEE_ENV.apigee.net/v1/oauth20/callback"
+    export TEST_IDP_APIGEE_REDIRECT_URI
+    
+    TEST_IDP_AZ_HOSTNAME=$(printf '%s' "$authorization_endpoint" | awk -F\"https:// '{print $2}' | awk -F\" '{print $1}' | awk -F/ '{print $1}')
+    export TEST_IDP_AZ_HOSTNAME
+    
+    TEST_IDP_TOKEN_HOSTNAME=$(printf '%s' "$token_endpoint" | awk -F\"https:// '{print $2}' | awk -F\" '{print $1}' | awk -F/ '{print $1}')
+    export TEST_IDP_TOKEN_HOSTNAME
+    
+    TEST_IDP_JWKS_HOSTNAME=$(printf '%s' "$jwks_uri" | awk -F\"https:// '{print $2}' | awk -F\" '{print $1}' | awk -F/ '{print $1}')
+    export TEST_IDP_JWKS_HOSTNAME
+    
+    TEST_IDP_USERINFO_HOSTNAME=$(printf '%s' "$userinfo_endpoint" | awk -F\"https://  '{print $2}' | awk -F\" '{print $1}' | awk -F/ '{print $1}')
+    export TEST_IDP_USERINFO_HOSTNAME
+    
+    TEST_IDP_TOKEN_URI=$(printf '%s' "$token_endpoint" | awk -F "$TEST_IDP_TOKEN_HOSTNAME"'/' '{print $2}' | awk -F\" '{print $1}')
+    export TEST_IDP_TOKEN_URI
+    
+    TEST_IDP_AZ_URI=$(printf '%s' "$authorization_endpoint" | awk -F "$TEST_IDP_AZ_HOSTNAME"'/' '{print $2}' | awk -F\" '{print $1}')
+    export TEST_IDP_AZ_URI
+    
+    TEST_IDP_JWKS_URI=$(printf '%s' "$jwks_uri" | awk -F "$TEST_IDP_JWKS_HOSTNAME"'/' '{print $2}' | awk -F\" '{print $1}')
+    export TEST_IDP_JWKS_URI
+    
+    TEST_IDP_USERINFO_URI=$(printf '%s' "$userinfo_endpoint" | awk -F "$TEST_IDP_USERINFO_HOSTNAME"'/' '{print $2}' | awk -F\" '{print $1}') 
+    export TEST_IDP_USERINFO_URI
+
+    TEST_IDP_APIGEE_CLIENT_ID="dummy-client_id-apigee123"
+    export TEST_IDP_APIGEE_CLIENT_ID
+
+    TEST_IDP_APIGEE_CLIENT_SECRET="dummy-client_secret_apigee456"
+    export TEST_IDP_APIGEE_CLIENT_SECRET
+}
+
+####################################
+### function: generate_edge_json ###
+####################################
+generate_edge_json()
+{
+  rm -f edge.json
+  cat <<EOF >> "edge.json"
```

Comment:
if it is easier for beginners to understand, we can hardcode in edge.json file at root instead of doing in pipeline

Reply:
I explained in the README that we generate a edge.json file containing the configuration of KVM, cache, product, dev app,...